server/enforcer.go

```diff
@@ -89,6 +98,11 @@ func (s *Server) NewAdapter(ctx context.Context, in *pb.NewAdapterRequest) (*pb.
 	return &pb.NewAdapterReply{Handler: int32(h)}, nil
 }
 
+type abacModel struct {
+	Name  string
+	Owner string
+}
```

Comment:
I think we need to be able to create Go structs from JSON on the fly without a predetermined structure since the attributes will be arbitrary in the general case.

Reply:
Otherwise, the evaluation function in Casbin itself could extract object/subject/resource attributes from JSON fields just like from Go structs. In other words, the `enforce` method should accept a JSON object as an input, not just a struct (for ABAC) or a string (for other policies)